lib/pages/chat/events/download_video_widget.dart

```diff
@@ -109,7 +109,9 @@ class _DownloadVideoWidgetState extends State<DownloadVideoWidget>
                   switch (downloadState) {
                     case DownloadVideoState.loading:
                       return InkWell(
-                        onTap: _downloadAction,
+                        onTap: downloadState == DownloadVideoState.loading
```

Comment:
this is inside `case DownloadVideoState.loading:`. Why we need this check?

Reply:
The issue occurs because clicking the download button multiple times during video download causes the app to initiate multiple downloads of the same video.